shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -67,15 +77,62 @@ fn rewrite_port(message: &mut Message, new_port: u32) {
         if let CassandraOperation::Result(CassandraResult::Rows { value, metadata }) =
             &mut frame.operation
         {
-            let port_column_index = metadata
-                .col_specs
+            let columns = ["native_port", "rpc_port", "broadcast_port", "peer_port"];
+
+            let port_column_indices = columns
+                .iter()
+                .map(|col| {
+                    metadata
+                        .col_specs
+                        .iter()
+                        .position(|wanted_col| &wanted_col.name.as_str() == col)
+                })
+                .flatten()
+                .collect::<Vec<_>>();
+
+            if !port_column_indices.is_empty() {
```

Comment:
This if statement is just for performance reasons right?
Can we have the same effect by just making `for i in &port_column_indice` the outermost loop?

Reply:
We can do this: 
```rust
if let MessageValue::Rows(rows) = &mut *value {
    for i in &port_column_indices {
        for row in rows.iter_mut() {
            row[*i] = MessageValue::Integer(new_port as i64, IntSize::I32);
        }
    }
    message.invalidate_cache();
}
```

But we can't move it to the outer-most for borrow checker reasons on `&mut *value`